python-mini-projects/projects/Fetch HTTP status code/fetch_http_status_code.py

```diff
@@ -0,0 +1,22 @@
+#Program to fetch the http status code give the url/api
+from urllib.request import urlopen
+from urllib.error import URLError, HTTPError
+import emoji
+
+#Taking input url from user
+requestURL = input("Enter the URL to be invoked: ")
+
+#Gets the response from URL and prints the status code, corresponding emoji and message accordingly
+try:
+    response = urlopen(requestURL)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *response*. The resource is allocated by call *request.urlopen*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *response* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).